<launch>
  <!-- Enable/disable and connect toggles per arm -->
  <arg name="enable_jaka1" default="true" />
  <arg name="enable_jaka2" default="false" />
  <arg name="enable_jaka3" default="false" />
  <arg name="enable_jaka4" default="false" />

  <arg name="connect_jaka1" default="true" />
  <arg name="connect_jaka2" default="true" />
  <arg name="connect_jaka3" default="true" />
  <arg name="connect_jaka4" default="true" />

  <arg name="control_rate_hz" default="20.0" />
  <arg name="kp_pos" default="0.6" />
  <arg name="kp_ang" default="0.8" />
  <arg name="v_max" default="0.02" />
  <arg name="w_max_deg" default="10.0" />
  <arg name="eps_pos_m" default="0.0015" />
  <arg name="eps_ang_deg" default="0.5" />
  <arg name="vision_timeout_s" default="0.2" />
  <arg name="servo_timeout_s" default="10.0" />

  <!-- jaka1 -->
  <group ns="jaka1" if="$(arg enable_jaka1)">
    <node pkg="jaka_close_contro" type="pose_servo_world_node" name="pose_servo_world" output="screen">
      <param name="robot_name" value="jaka1" />
      <param name="robot_id" value="1" />
      <param name="enable" value="$(arg enable_jaka1)" />
      <param name="connect_robot" value="$(arg connect_jaka1)" />
      <param name="calib_yaml" value="$(find jaka_close_contro)/config/world_robot_extrinsic_offline_jaka1.yaml" />
      <param name="cube_pose_topic" value="/vision/jaka1/cube_center_world" />
      <param name="target_topic" value="target_tool_world" />
      <param name="control_rate_hz" value="$(arg control_rate_hz)" />
      <param name="kp_pos" value="$(arg kp_pos)" />
      <param name="kp_ang" value="$(arg kp_ang)" />
      <param name="v_max" value="$(arg v_max)" />
      <param name="w_max_deg" value="$(arg w_max_deg)" />
      <param name="eps_pos_m" value="$(arg eps_pos_m)" />
      <param name="eps_ang_deg" value="$(arg eps_ang_deg)" />
      <param name="vision_timeout_s" value="$(arg vision_timeout_s)" />
      <param name="servo_timeout_s" value="$(arg servo_timeout_s)" />
    </node>
  </group>

  <!-- jaka2 -->
  <group ns="jaka2" if="$(arg enable_jaka2)">
    <node pkg="jaka_close_contro" type="pose_servo_world_node" name="pose_servo_world" output="screen">
      <param name="robot_name" value="jaka2" />
      <param name="robot_id" value="2" />
      <param name="enable" value="$(arg enable_jaka2)" />
      <param name="connect_robot" value="$(arg connect_jaka2)" />
      <param name="calib_yaml" value="$(find jaka_close_contro)/config/world_robot_extrinsic_offline_jaka2.yaml" />
      <param name="cube_pose_topic" value="/vision/jaka2/cube_center_world" />
      <param name="target_topic" value="target_tool_world" />
      <param name="control_rate_hz" value="$(arg control_rate_hz)" />
      <param name="kp_pos" value="$(arg kp_pos)" />
      <param name="kp_ang" value="$(arg kp_ang)" />
      <param name="v_max" value="$(arg v_max)" />
      <param name="w_max_deg" value="$(arg w_max_deg)" />
      <param name="eps_pos_m" value="$(arg eps_pos_m)" />
      <param name="eps_ang_deg" value="$(arg eps_ang_deg)" />
      <param name="vision_timeout_s" value="$(arg vision_timeout_s)" />
      <param name="servo_timeout_s" value="$(arg servo_timeout_s)" />
    </node>
  </group>

  <!-- jaka3 -->
  <group ns="jaka3" if="$(arg enable_jaka3)">
    <node pkg="jaka_close_contro" type="pose_servo_world_node" name="pose_servo_world" output="screen">
      <param name="robot_name" value="jaka3" />
      <param name="robot_id" value="3" />
      <param name="enable" value="$(arg enable_jaka3)" />
      <param name="connect_robot" value="$(arg connect_jaka3)" />
      <param name="calib_yaml" value="$(find jaka_close_contro)/config/world_robot_extrinsic_offline_jaka3.yaml" />
      <param name="cube_pose_topic" value="/vision/jaka3/cube_center_world" />
      <param name="target_topic" value="target_tool_world" />
      <param name="control_rate_hz" value="$(arg control_rate_hz)" />
      <param name="kp_pos" value="$(arg kp_pos)" />
      <param name="kp_ang" value="$(arg kp_ang)" />
      <param name="v_max" value="$(arg v_max)" />
      <param name="w_max_deg" value="$(arg w_max_deg)" />
      <param name="eps_pos_m" value="$(arg eps_pos_m)" />
      <param name="eps_ang_deg" value="$(arg eps_ang_deg)" />
      <param name="vision_timeout_s" value="$(arg vision_timeout_s)" />
      <param name="servo_timeout_s" value="$(arg servo_timeout_s)" />
    </node>
  </group>

  <!-- jaka4 -->
  <group ns="jaka4" if="$(arg enable_jaka4)">
    <node pkg="jaka_close_contro" type="pose_servo_world_node" name="pose_servo_world" output="screen">
      <param name="robot_name" value="jaka4" />
      <param name="robot_id" value="4" />
      <param name="enable" value="$(arg enable_jaka4)" />
      <param name="connect_robot" value="$(arg connect_jaka4)" />
      <param name="calib_yaml" value="$(find jaka_close_contro)/config/world_robot_extrinsic_offline_jaka4.yaml" />
      <param name="cube_pose_topic" value="/vision/jaka4/cube_center_world" />
      <param name="target_topic" value="target_tool_world" />
      <param name="control_rate_hz" value="$(arg control_rate_hz)" />
      <param name="kp_pos" value="$(arg kp_pos)" />
      <param name="kp_ang" value="$(arg kp_ang)" />
      <param name="v_max" value="$(arg v_max)" />
      <param name="w_max_deg" value="$(arg w_max_deg)" />
      <param name="eps_pos_m" value="$(arg eps_pos_m)" />
      <param name="eps_ang_deg" value="$(arg eps_ang_deg)" />
      <param name="vision_timeout_s" value="$(arg vision_timeout_s)" />
      <param name="servo_timeout_s" value="$(arg servo_timeout_s)" />
    </node>
  </group>

</launch>
